docs/vc-http-api.yml

```
@@ -116,6 +138,24 @@ paths:
 
 components:
   schemas:
+    UpdateCredentialStatus:
+      type: object
+      description: Request for updating the status of an issued credential.
+      oneOf:
+        - type: object
+          properties:
+            credentialId:
+              type: string
+            type:
+              type: string
+            status:
+              type: number
+      example:
+        { 
+          "credentialId": "urn:uuid:45a44711-e457-4fa8-9b89-69fe0287c86a",
+          "type": "RevocationList2020Status",
+          "status": 0
```

```suggestion
          "id": "urn:uuid:45a44711-e457-4fa8-9b89-69fe0287c86a",
          "credentialStatus": [{
              "type": "RevocationList2020Status",
              "statusValue": 0
           }]
```

I'm concerned about both of these proposals for the object.

@dlongley, thoughts?

---

@msporny agree with the functionality you are trying to enable, but how do we unambiguously refer to the different credential statuses in the array you have defined? I.e are we assuming that a credential will never be issued with more than one credential status of the same type?

---

Another approach would be to have this endpoint only update a single status associated to the credential and require the requestor to make multiple requests to update all the credential statuses?

---

> are we assuming that a credential will never be issued with more than one credential status of the same type?

I believe so -- unless someone has a use case where they are publishing two revocation lists associated with the same credential.

> Another approach would be to have this endpoint only update a single status associated to the credential and require the requestor to make multiple requests to update all the credential statuses?

I don't think we have to do that... the current form allows you to update all credential statuses associated with the VC in one shot.

---

> I believe so -- unless someone has a use case where they are publishing two revocation lists associated with the same credential.

Ok, I agree at this stage it would be a strange application just pointing out that this mechanism for identifying the associated credential status to update is not 100% explicit and also that it is valid from the VC data model perspective to have multiple statues of the same type in a single credential.

> Unless I'm missing something, I don't think we have to do that... the current form allows you to update all credential statuses associated with the VC in one shot.

Agree im not against this style of bulk/multi status update just calling that out as our other option.